src/main/java/pl/hk/zadanie_31/controller/HomeController.java

```diff
@@ -0,0 +1,39 @@
+package pl.hk.zadanie_31.controller;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import pl.hk.zadanie_31.model.Weather;
+import pl.hk.zadanie_31.service.WeatherService;
+
+@Controller
+public class HomeController {
+
+    private final WeatherService weatherService;
+
+    public HomeController(WeatherService weatherService) {
+        this.weatherService = weatherService;
+    }
+
+    @GetMapping("/")
+    public String home(Model model, WeatherDto weather, @ModelAttribute("data") WeatherDto weatherDto){
```

Comment:
W argumentach masz dwa razy WeatherDto. Powinno być tylko raz i z tego co widzę, to chciałbyś zostawić to z adnotacją @ModelAttribute